src/app/icd/client/CheckInDelegate.h

```diff
@@ -0,0 +1,33 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+namespace chip {
+namespace app {
+
+/// Callbacks for check in protocol
+class DLL_EXPORT CheckInDelegate
+{
+public:
+    virtual ~CheckInDelegate() {}
+    virtual void OnCheckInComplete(void) = 0;
```

Comment:
Please add PeerNodeId for input parameter

Reply:
virtual void OnCheckInComplete(const ScopedNodeId & peerNodeId, bool needRegisterICD) = 0; we may consider to ask application to registerICD